### ISSUE TITLE
Toward Controlled Generation of Text

### ISSUE THREAD

**Issue by noowad** [noowad/paper-summaries#19]
## 1. 논문 정보
ICML 2017
Toward Controlled Generation of Text
Zhiting Hu, Zichao Yang, Xiaodan Liang, Ruslan Salakhutdinov, Eric P Xing
Carnegie Mellon University
## 2. 개요
VAE와 attribute discriminators를 이용하여 특정 속성을 갖는 문장을 생성하는 수법에 관한 연구이다. 본 논문에서는 Sentiment와 Tense에 관한 데이터셋을 이용하여 속성을 변환하는 실험을 통해 수법을 평가하였다. 
## 3. 선행 연구보다 발전된 점
텍스트 생성에 있어서 크게 2가지 과제가 존재한다.
1. 출력 결과가 텍스트 (이산적인 값)이므로, 미분이 불가능하기 때문에 식별기로부터의 오차를 backprop할 수 없다.
2. 제어가능한 문장을 생성하기 위한 복잡한 latent representations을 학습하기가 어렵다.

본 연구에서는 2가지 문제에 대한 해결책을 제안했고, 어느 정도 제어가 가능한 문장을 생성함을 보였다.
## 4. 어떠한 수법을 사용하였는가?
본 연구의 텍스트 생성 수법은 다음과 같다.
![2018-07-16 0 49 04](https://user-images.githubusercontent.com/26961984/42735613-1d1f111c-8892-11e8-934f-6b89009f065f.png)
VAE는 생성기가 타당한 문장을 생성할 수 있도록 훈련되고, 식별기는 생성기가 지정한 변수에 맞는 속성을 갖는 문장을 생성하도록 제어한다.
본 연구의 VAEs는 구조화된 변수 c (속성)와 구조화되지 않은 z를 조건으로하여 출력 문장을 제어한다. 여기서 제어하고 싶은 속성은 c를 통하여 독립적으로 입력받고, 그 외의 내용 부분은 z를 통하여 정보를 제공받는다. 또한, 식별기로부터의 backprop을 위하여 softmax with a decreasing temperature를 이용한다.
알고리즘은 다음과 같다.
![2018-07-16 1 02 34](https://user-images.githubusercontent.com/26961984/42735717-00131f58-8894-11e8-915b-eaee056145d5.png)
## 5. 검증
감정과 시제가 제어된 짧은 문장을 생성하는 실험을 통해 수법을 검증했다.
데이터셋으로는 
1. 문장 코퍼스
IMDB의 텍스트 코퍼스 (350K) 중 15단어 이하의 문장을 선택하여 학습에 이용하였다.
2. 감정
텍스트의 감정을 제어하기 위해 4개의 라벨이 붙어있는 데이터셋으로 테스트하였다. 데이터셋으로는 Stanford Sentiment Treebank-2 (SST-full)과 SST-small, Lexicon, IMDB의 코퍼스를 이용하였다.
3. 시제
TimeBank의 데이터셋을 이용하여 과거,현재,미래의 라벨이 붙어있는 약 5000개의 단어와 구를 이용하였다. 

감정에 관한 정확도 실험 결과는 다음과 같다. 
![2018-07-16 1 04 54](https://user-images.githubusercontent.com/26961984/42735773-ec87fade-8894-11e8-9ac4-b362fe53f6e9.png)
![2018-07-16 1 12 39](https://user-images.githubusercontent.com/26961984/42735801-5bfd4b8a-8895-11e8-8499-933dd472302e.png)
왜인지 모르겠으나, Lexicon의 결과가 SST-full보다 낮게 나온다.

시제를 제어한 문장의 예는 다음과 같다.
![2018-07-16 1 13 02](https://user-images.githubusercontent.com/26961984/42735803-743a6d86-8895-11e8-90d9-f2fcc49f15eb.png)

## 6. 개인적인 감상
우선 실험을 할 때, 15단어 미만의 짧은 문장만을 이용했고 cherry picking을 했기 때문에, 논문에 제시된 예시 자체는 나쁘지 않음. 평가 항목으로써 특정 속성을 담고 있는지에 대한 부분만 평가를 했기 때문에 속성 이외의 내용 보존 및 문법성 등에 대해서도 평가할 필요가 있다고 생각한다. 최근 생성 모델을 이용한 문장 생성도 여러 수법을 통해 진행되고 있고 여러가지 과제들이 해결되어가는 느낌이라, 1~2년 사이에 image 분야 수준의 생성 수법이 제안되지 않을까 싶다.
## 7. 다음에 읽어 볼 논문
